src/database.rs

```diff
@@ -268,25 +268,25 @@ impl Database {
                         {
                             let mut keydir = keydir.lock().unwrap();
                             let set_result = keydir.set(&entry.key, entry.file_id, entry.offset, entry.timestamp);
-                            if set_result.is_err() {
-                                trace!("Setting value into keydir has failed: {}", set_result.err().unwrap());
+                            if let Err(err) = set_result {
+                                trace!("Setting value into keydir has failed: {}", err);
```

Comment:
`if let Err(err) = some_res {}` unwraps the Result if its an Err and its available inside of the body